docs/tutorials/auth.md

```diff
@@ -415,3 +415,90 @@ scope by doing the following:
 
 > The subject is the field that identifies the user and is often called `sub`, 
 > especially in the context of OpenID Connect. In the example above, it is `username`.
+
+## Open Policy Agent
+
+To enable [Open Policy Agent](https://www.openpolicyagent.org/) filter, use the `-enable-open-policy-agent` command line flag. 
+
+Open Policy Agent is integrated as a Go library so no extra setup is needed to run. Every filter creates a virtual OPA instance in memory that is configured using a configuration file in the same [configuration format](https://www.openpolicyagent.org/docs/latest/configuration/) that a standalone OPA would use. To allow for configurability, the configuration file is interpolated using Go Templates to allow every virtual instance to pull different bundles. This template file is passed using the `-open-policy-agent-config-template` flag. 
+
+### Configuration File
+
+As an example the following initial config can be used
+
+```yaml
+services:
+  - name: bundle-service
+    url: https://my-example-opa-bucket.s3.eu-central-1.amazonaws.com
+    credentials:
+      s3_signing:
+        environment_credentials: {}
+labels:
+  environment: production
+discovery:
+  name: discovery
+  prefix: "/applications/{{ .bundlename }}"
+```
+
+The variable `.bundlename` is the first argument in the following filters and can be in any format that OPA can understand, so for example application IDs from a registry, uuids, ...
+
+### Envoy Structures
+
+While Envoy is an alternative OSS product similar to Skipper, it has already defined structures for how external authorization should be done and also how  authorization decisions can influence the Envoy response. Open Policy Agent already has direct support for this and also commercial control planes support this. On top of this [examples and documentation](https://www.openpolicyagent.org/docs/latest/envoy-primer/) already exist.
```

Comment:
`how  authorization` -> `how authorization`

Reply:
Reading that everyone uses this data structure, maybe we should not name it "envoy", but have one reference that we use the same as envoy as everyone else. We don't really need to "pollute" the names with envoy all over the place?!

Reply:
Agreed. The text was for a different audience, will shorten it to one sentence that references the existing docs.